Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -18,15 +18,31 @@ import StoreKit
 @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
 class PurchasedProductsFetcher {
 
+    private typealias Transactions = [StoreKit.VerificationResult<StoreKit.Transaction>]
+
     private let appStoreSync: () async throws -> Void
     private let sandboxDetector: SandboxEnvironmentDetector
+    private let cache: InMemoryCachedObject<Transactions>
```

Comment:
So you are planning to cache the transactions instead of the whole offline customer info? While that should probably solve the performance concerns it might not help to support fetching customer info from cache while in offline entitlements mode, unless we add support for recalculating it there.